language_service/src/rename.rs

```diff
@@ -0,0 +1,375 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#[cfg(test)]
+mod tests;
+
+use std::rc::Rc;
+
+use qsc::{
+    ast::{
+        self,
+        visit::{walk_callable_decl, walk_expr, walk_pat, walk_ty, walk_ty_def, Visitor},
+    },
+    hir::{self, ty::Ty, Res},
+    resolve, Span,
+};
+
+use crate::{
+    protocol,
+    qsc_utils::{map_offset, Compilation},
+};
+
+pub(crate) fn prepare_rename(
+    compilation: &Compilation,
+    source_name: &str,
+    offset: u32,
+) -> Option<(protocol::Span, String)> {
+    // Map the file offset into a SourceMap offset
+    let offset = map_offset(&compilation.unit.sources, source_name, offset);
+    let package = &compilation.unit.ast.package;
+
+    let mut prepare_rename = Rename::new(compilation, offset, true);
+    prepare_rename.visit_package(package);
+    prepare_rename.prepare.map(|p| {
+        (
+            protocol::Span {
+                start: p.0.lo,
+                end: p.0.hi,
+            },
```

Comment:
I think it's worth creating a tiny helper function for this (and the same pattern below in the lambda inside `get_rename`). 

Also, do we map `SourceMap` offsets to source offsets somewhere, like [`protocol_span` in hover.rs](https://github.com/microsoft/qsharp/blob/598b939260c6da90e0d8b66ffda35bd94c51aee2/language_service/src/hover.rs#L310C16-L310C16)?

Reply:
I hadn't thought to do the source mapping for this because there's no other files to test against yet, but it is a good idea to add it now to make it future compatible with multi-file. I've added a helper to the utils that does this mapping.